### ISSUE TITLE
SBT 0.13.7 does not react on Ctrl-C or Ctrl-D hotkeys

### ISSUE THREAD

**Issue by tttomat19** [sbt/sbt#1855]
### Steps to reproduce:

Win 8.1 ruRU, chcp 1251 (866, 60001 and 1252 don't work either)
build.sbt https://gist.github.com/flirtomania/e1273618ab706a39eaba
Main class: https://gist.github.com/flirtomania/aee23feca65126db6c61
1. run sbt in project directory (now it does not react on ctrl-c, still terminates on ctrl-d)
2. call "run" task, (application runs ok)
### Problems:

sbt run task does not react on ctrl-c or ctrl-d hotkeys
### Expectations:

pressing ctrl-c / ctrl-d should cause run task to stop or quit sbt

**Comment by eed3si9n** [comment]
Hi, could you see #1442 and see if 

``` scala
cancelable in Global := true
```

fixes it? Also is this an Akka specific issue?

**Comment by tttomat19** [comment]
Hi, `cancelable in Global := true` didn't work.
The issue reproduces also on Windows 7 running with `sbt -Dinput.encoding=Cp1252` and without input encoding also (this is related to up arrow behaviour on russian versions of windows).
This is how I tried without Akka: https://gist.github.com/flirtomania/a2c291551c5b53c7e1f4
It still did not react on Ctrl-C or Ctrl-D
Both Windows 8.1 and Windows 7 have Russian language. (Cp866)

**Comment by torgeirsh** [comment]
Ctrl-C doesn't work for me either, with or without "cancleable" set. Ctrl-D does work however (Windows 8.1).

**Comment by mkotsbak** [comment]
Same problem running a Spring Boot application in Windows with "sbt run". "cancelable in Global := true" does not help. Using sbt 0.13.8.

**Comment by mkotsbak** [comment]
I noticed now that it is not stoppable when running "run" from within the sbt console, but when running "sbt run" directly from the windows terminal/cygwin, ctrl-c works fine.

**Comment by Araeos** [comment]
I also experience this issue on windows 10 with sbt 0.13.9 and it hinders the workflow when a task, such as run, cannot be interrupted without killing the whole cmd/powershell session.
Additionally, when forked is set to true, the forked task will run even after sbt has been killed.

**Comment by aholland** [comment]
Still a problem on Windows 10 with `sbt ~run` running `sbt 0.13.11`

**Comment by nartamonov** [comment]
Same story with Windows 7 and SBT 0.13.8. Since I started working with sbt in 2012 I can't remember that Ctrl+C works as expected. It's one of many small things that makes work with SBT harassing :weary:

**Comment by folex** [comment]
Same thing with sbt 0.13.8 -- Ctrl-C doesn't work either

**Comment by frozenspider** [comment]
Pretty annoying, any updates on this?

**Comment by Araeos** [comment]
This seems to be a bug with jline, because using version 2.14.3 fixes this issue for me. The relevant issue jline/jline2#127 mentions as much.
Since the currently stable release does not use jline version 2.14.3 (or above) I just replaced the `jline.jar` in my .sbt folder (`C:\Users\<USER>\.sbt\boot\scala-2.10.6\lib\jline.jar`) with a newer version from the [maven repository](https://mvnrepository.com/artifact/jline/jline). This is just a dirty hack and since the original file was likely jline version 2.13 this might cause severe bugs due to incompatibilies.
Also, the jline version seems to be a dependency of scala-2.10.6 and might not be easily updated with just an sbt release.
I hope this clears things up as to why ctrl+c is ignored completely by sbt on windows.

**Comment by dwijnand** [comment]
TL;DR: We should update JLine. At least 2.14.3. The Scala Team has already.

Thanks @Araeos for the detail and cross-linking.

**Comment by eed3si9n** [comment]
This issue will now be expedited via the Lightbend subscription.

**Comment by ChuckkNorris** [comment]
The latest advertised release linked on the Download Scala states it's for Scala v2.12.3, but it downloads the the SBT v1.0.2 which still installs Scala v2.10.6 (with the bad `jline.jar`).

Specifically, this is occurring for the Windows MSI installer. Any idea when the link will be updated so new people don't have to deal with this error?
